### ISSUE TITLE
coadd- files, total exposure time information

### ISSUE THREAD

**Issue by segasai** [desihub/desispec#1078]
Hi, 

while analyzing latest outputs of the data processing pipeline I've realized that I can easily figure out the exposure time for each non-stacked spectra through lookup by EXPID for each spectra in spectra-*fits file , but for stacks It doesn't seem easily possible, as (to my knowledge) only the FIRST/LAST EXPID and number of exposures is stored in coadd-* files. So I was wondering if it would be good to store total EXPTIME in the FIBERMAP table of the coadd-*  files ? 

If yes, I can probably provide a PR. 
And in this case it's probably good to store EXPTIME in spectra-*files as well. 

Or am I missing some other easy way of extracting total exptime info  ?

Thanks, 
        Sergey

**Comment by sbailey** [comment]
This is indeed a pain and a PR along the lines of what you describe would be welcome.  Currently I think your only option is:
1. coadd*.fits HDU 0 has keywords INFILnnn listing the input files for the coadd, which could be one or multiple spectra* or cframe* files.
2. Loop over those INFILnnn entries to look at the NIGHT, EXPID in the FIBERMAP HDU, from which you can lookup the original raw data or input cframe file to get the EXPTIME from the header or via a DB query (sounds like you are doing something similar to this from the full spectra files)

**Comment by segasai** [comment]
I've created a first iteration PR #1079 to deal with this. I add EXPTIME to the FIBERMAP of single exposures and COADD_EXPTIME to the fibermap of the stack. 
I'm guessing a common name of EXPTIME is probably better. I can change that.
It is also a bit weird to included a column with the same value in the FIBERMAP  (ideally that'd be better stored in the header), but I think the convenience outweight normalization reasoning.
I hope I didn't miss anything obvious in the patch.

      S
(And previously I did indeed just create my own database of EXPID/EXPTIME map by just scanning all the exposure images)
